zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/DeleteRoleMultiPartitionTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.authorization;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.tuple;
+
+import io.camunda.zeebe.engine.state.distribution.DistributionQueue;
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.AuthorizationIntent;
+import io.camunda.zeebe.protocol.record.intent.CommandDistributionIntent;
+import io.camunda.zeebe.protocol.record.intent.RoleIntent;
+import io.camunda.zeebe.protocol.record.value.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.CommandDistributionRecordValue;
+import io.camunda.zeebe.protocol.record.value.PermissionAction;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.time.Duration;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class DeleteRoleMultiPartitionTest {
+
+  private static final int PARTITION_COUNT = 3;
+
+  @Rule public final EngineRule engine = EngineRule.multiplePartition(PARTITION_COUNT);
+
+  @Rule public final TestWatcher testWatcher = new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldDistributeRoleDeleteCommand() {
+    // when
+    final var name = UUID.randomUUID().toString();
+    final var roleKey = engine.role().newRole(name).create().getValue().getRoleKey();
+    engine
+        .authorization()
+        .permission()
+        .withOwnerKey(roleKey)
+        .withOwnerType(AuthorizationOwnerType.ROLE)
+        .withResourceType(AuthorizationResourceType.ROLE)
+        .withAction(PermissionAction.REMOVE)
+        .add();
+    engine.role().deleteRole(roleKey).withName(name).delete();
```

Comment:
❌ Let's remove the `withName` part. We shouldn't need to provide a name upon deleting a Role.

Reply:
At the moment the name is mandatory, because in the `delete` in the role state we are removing also the role from the column name. 
We can retrieve this information from the state in the processor, I wasn't sure about this approach tho

Reply:
Oh I see where it went wrong. In the processor you are getting the role by the key. You can use this to set the name on the record. 

`record.setName(persistedRole.getName());`

Then you no longer need to provide the name in the tests. We need to change this, because in the REST API we won't get anything but a key, so we need to get the name ourselves.